extmod/moductypes.c

```diff
@@ -107,6 +107,9 @@ enum {
 // We cannot apply the below to INT8, as their range [-128, 127]
 #define IS_SCALAR_ARRAY_OF_BYTES(tuple_desc) (GET_TYPE(MP_OBJ_SMALL_INT_VALUE((tuple_desc)->items[1]), VAL_TYPE_BITS) == UINT8)
 
+#define IS_ALIGNED(addr, val_type) (!(((uintptr_t)addr) & (GET_SCALAR_SIZE(val_type)-1)))
```

Comment:
I don't think this having this macro is needed/a correct way. The rules are simple:if structure is native, you must use integral instructions to access values of given size (i.e., access 8, 16, 32, 64 bit quantities as such). Whether it's aligned or not is not important on this level - it's up to the designer of the structure, who should keep in mind that on many architecture, unaligned access to integral values lead to exceptions (so, if someone uses unaligned native structure on such CPU, exceptions is what they want, and thus, must get).

When using non-native structures, there's no constraint of integral access, any quantity may be accessed byte by byte, and alignment exceptions must not happen.